git-publish

```diff
@@ -884,9 +902,13 @@ branch.%s.pushRemote is set appropriately?  (Override with --no-check-url)''' %
                 edit(*patches)
             if cc_cmd:
                 for x in patches:
-                    output = subprocess.check_output(cc_cmd + " " + x,
-                                shell=True, cwd=git_get_toplevel_dir()).decode(ENCODING)
-                    cc = cc.union(output.splitlines())
+                    # Unless we provide the encoding paramter for cc_cmd, otherwise should use git default
```

Comment:
I reviewed git's `git-send-email.perl` to understand the encoding that cc_cmd has to produce: it's the locale's encoding (often utf-8 but it may be something else) according to https://perldoc.perl.org/5.34.1/perluniintro#Perl's-Unicode-Model. GIT_ENCODING will work most of the time but I think Python's default encoding might be correct when the user has a non-UTF-8 locale.

I bet a lot of cc-cmd scripts don't get the encoding right though. Let's try it with the git default but please update the comment to say "The encoding of cc-cmd output is not well-defined. Use git's encoding for now although git-send-email is a Perl script that uses Perl's Unicode support rather than git's standard UTF-8 encoding.".